src/output/plugins/OSXOutputPlugin.cxx

```diff
@@ -688,7 +680,44 @@ osx_render(void *vdata,
 {
 	OSXOutput *od = (OSXOutput *) vdata;
 
-	int count = in_number_frames * od->asbd.mBytesPerFrame;
+	if (od->cancel) {
+		od->ring_buffer->reset();
+		od->cancel = false;
+	}
+
+	// If data is not available, fill the buffer with silence,
```

Comment:
Can we move the silence generation code in a separate commit? Mixing two orthogonal changes in one commit may make bisects harder than necessary, and obscures the patch.

Reply:
https://github.com/MusicPlayerDaemon/MPD/pull/598 does not contain this change. Will do in another PR.